Assets/Scripts/LevelBuilder.cs

```diff
@@ -0,0 +1,88 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class LevelBuilder : MonoBehaviour
+{
+    public GameObject[] SafePrefabs;
+    public GameObject[] DangerPrefabs;
+
+    public GameObject[] backGrounds;
+
+    public AnimationCurve SafeChance;
+    public float MaxHeight = 100f;
+
+    public float minStep = 2f, maxStep = 8f;
+
+    public float maxShift = 12f;
+
+    [SerializeField] GameObject lastPlaced;
+    GameObject lastGenerated;
+    [SerializeField] float levelWidth = 5.0f;
+
+    List<GameObject> clouds = new List<GameObject>();
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        GetWidth();
+        lastGenerated = lastPlaced;
+    }
+
+    // Update is called once per frame
+    void Update()
+    {   
+        if (lastGenerated.transform.position.y < Camera.main.transform.position.y + Camera.main.orthographicSize + maxStep)
+            clouds.Add(PlaceNewCloud());
+        for(int i = clouds.Count - 1; i >= 0; i--)
```

Comment:
For a second there, I was confused as to why you were doing the (""; ""; i--) but as i read further down I got it, what you're doing here is removing the cloud after the camera has panned up.

Reply:
Yes! And looping backward through the list means we can remove items from it without modifying the index of remaining items